File: app/src/main/java/co/aspirasoft/sams/ActivitySplash.java
package co.aspirasoft.sams;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.os.CountDownTimer;
import androidx.appcompat.app.AppCompatActivity;

public class ActivitySplash extends AppCompatActivity {

    public static final String PREFS_FILE = "PreferencesFile";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        new CountDownTimer(1500, 10) {

            @Override
            public void onTick(long millisUntilFinished) {
            }

            @Override
            public void onFinish() {
                SharedPreferences sharedPreferences = getSharedPreferences(PREFS_FILE, 0);
                Boolean userKnown = sharedPreferences.getBoolean("isActiveSession", false);

                if (userKnown)
                    startActivity(new Intent(getApplicationContext(), ActivitySchedule.class));
                else
                    startActivity(new Intent(getApplicationContext(), ActivitySignIn.class));

                finish();
            }
        }.start();
    }
}

File: app/src/main/java/co/aspirasoft/sams/ui/activity/ActivitySplash.kt
package co.aspirasoft.sams.ui.activity

import android.content.Intent
import android.os.Bundle
import android.os.CountDownTimer
import androidx.appcompat.app.AppCompatActivity
import co.aspirasoft.sams.R

class ActivitySplash : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)

        object : CountDownTimer(1500, 10) {
            override fun onTick(millisUntilFinished: Long) {}
            override fun onFinish() {
                val sharedPreferences = getSharedPreferences(PREFS_FILE, 0)
                val userKnown = sharedPreferences.getBoolean("isActiveSession", false)
                if (userKnown) startActivity(
                    Intent(
                        applicationContext,
                        ActivitySchedule::class.java
                    )
                ) else startActivity(
                    Intent(
                        applicationContext, ActivitySignIn::class.java
                    )
                )
                finish()
            }
        }.start()
    }

    companion object {
        const val PREFS_FILE = "PreferencesFile"
    }

}